apps/web/modules/auth/signup/lib/invite.ts

```diff
@@ -0,0 +1,79 @@
+import { inviteCache } from "@/lib/cache/invite";
+import { InviteWithCreator } from "@/modules/auth/signup/types/invites";
+import { Prisma } from "@prisma/client";
+import { cache as reactCache } from "react";
+import { prisma } from "@formbricks/database";
+import { cache } from "@formbricks/lib/cache";
+import { validateInputs } from "@formbricks/lib/utils/validate";
+import { ZString } from "@formbricks/types/common";
+import { DatabaseError, ResourceNotFoundError } from "@formbricks/types/errors";
+
+export const deleteInvite = async (inviteId: string): Promise<boolean> => {
+  validateInputs([inviteId, ZString]);
+
+  try {
+    const invite = await prisma.invite.delete({
+      where: {
+        id: inviteId,
+      },
+      select: {
+        id: true,
+        organizationId: true,
+      },
+    });
+
+    if (!invite) {
+      throw new ResourceNotFoundError("Invite", inviteId);
+    }
+
+    inviteCache.revalidate({
+      id: invite.id,
+      organizationId: invite.organizationId,
+    });
+
+    return true;
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      throw new DatabaseError(error.message);
+    }
+
+    throw error;
+  }
+};
+
+export const getInvite = reactCache(
+  async (inviteId: string): Promise<InviteWithCreator | null> =>
+    cache(
+      async () => {
+        validateInputs([inviteId, ZString]);
+
+        try {
+          const invite = await prisma.invite.findUnique({
+            where: {
+              id: inviteId,
+            },
+            include: {
+              creator: {
+                select: {
+                  name: true,
+                  email: true,
+                },
+              },
+            },
+          });
+
+          return invite;
+        } catch (error) {
+          if (error instanceof Prisma.PrismaClientKnownRequestError) {
+            throw new DatabaseError(error.message);
+          }
+
+          throw error;
+        }
+      },
+      [`getInvite-${inviteId}`],
```

Comment:
we don't use the cache keys for now since next js doesnt have a function for this, but in general even for the same service, the cache keys should ideally be unique, also the prisma select statement can vary with each service even if all of them are called `getInvite`, this key `getInvite-${inviteId}` is being used for all the `getInvite` services. Please use unique keys for all the `getInvite` services like the `utils-getInvite-${inviteId}` cache key which is in `apps/web/lib/utils/services.ts`